File: benchmark/csharp/Services/XmlService.cs
using System.Xml;

namespace Benchmark.CSharp.Services
{
    public class XmlService
    {
        public void ParseXmlData(string xmlInput)
        {
            // XML External Entity (XXE) [CWE-611]
            XmlReaderSettings settings = new XmlReaderSettings();
            settings.DtdProcessing = DtdProcessing.Parse; 
            
            using (StringReader sr = new StringReader(xmlInput))
            using (XmlReader reader = XmlReader.Create(sr, settings))
            {
                XmlDocument doc = new XmlDocument();
                doc.Load(reader);
            }
        }
    }
}

File: benchmark/ruby/services/xml_service.rb
require 'nokogiri'

class XmlService
  def self.parse_xml(xml_data)
    # XML External Entity [CWE-611]
    doc = Nokogiri::XML(xml_data) do |config|
      config.noent
    end
    doc.to_s
  end
end
